src/Fantomas.Core/CodePrinter.fs

```diff
@@ -1834,15 +1828,36 @@ let genMultilineFunctionApplicationArguments (argExpr: Expr) =
     | _ -> genExpr argExpr
 
 let genTuple (node: ExprTupleNode) =
+    let wrapInfixAppRhsInParenIfNeeded expr =
+        match expr with
```

Comment:
Can you check what happens for `SameInfixApps`?

Reply:
They don't seem to be affected by the issue
See: [here](https://fsprojects.github.io/fantomas-tools/#/fantomas/main?data=N4KABGBEDGD2AmBTSAuKAbRAXMBDMAvGAAwA6AdpjgEaFhYBOArohVWNHQIxvZjx0ATBV44A%2BoQoQIjAJ5TpixizAAfAHxgAlgI3bdmnWsPwANGGoKIAdy1YAFmEQAPMAFp1VxWABmudADOiOZ%2Bgcim4FBBWFha5ADmAahgANqREKDeUAAkWLIADshokHFYkBFZkNkAbv4syWlZEMQVTZAAkuRI5FgAyloAXuHp3gAsIxAAuiMAvq0ZEzl5hcklPeWLVbXo9WiNTVzz3pAAsrjOADJxiBeICQ4bTWBcgsSL04pzI5nHuQVFUDu8AA8j4ruREL08phHr9trtUosIIIjopIABRLqg8HDJ6QdA%2BSDvWZHH5ov4rYrUWCwdCw8nwgH7LIAZlR0g65CCDCwADE4v4AHKIazoa707zKRDEz6kzYUgGQam0iUQLZ1JlIsCjdlq3r5XDQRAAIUQPlgDEQAAVcAxcABbbCIBiq6RSmXSL6KMkchWrZV09nqnaap4AVl1UH1hpNZotN1g1md0FwQU61VgKdisHIrpkzGlWQ%2BnrllT9VJpgfljIaWoAbJHINGjabzZaAKr5QoMFNp8gZrNaHN53z%2BIIeiBe6Q%2BtXlqAB13BhHM7wAdkbzdjbcQAGF0KmAjucwFlNAsBaR6Fx0WSd95ctFQugzUNbWngAODcGltxy0nRD2tQzqXmOhbeMWk6lr8D7%2BpWi4viGb5NAAnF%2BMatvGR7oMOkZXmBigQWAU4LGWMEViqz41nsWpcC0WpNt%2BiAAII%2BFgzpHva9q4CO7o3rKd6kf8sEUdWr7UU8XCHPRm4YZavQAVocDYbmuGgRORFQeSZHznBlFiYiEkotJjEsWxDDyfaim0jhWq8eBt7eveQnkVWZZUQZBxssZMZMQwsBMF0AAiiBipZZk8QW6nEWAM5LM5UClPB7krooXA6vRZzOO0PgACr2HcvT2BaWAAOo6A8kZvHxJYCdB8VrGUemIeJBwRhl5zZXldzomEhXFWV8AVfWVX2fxjmCZSCXrE1y40Q27VZeQPhaM4wLdrg54MOizj5JaAQBEOKlau%2BI0EQ505OZNDVJfpKXSFw64LQASogcAMPAA1DU8oyndIhHRbFVTaddM2hgcn7Pa9FrwIKTCAc6oLtGx9pJJGXBRZpvrA-aTDoLEYoQryFpcTEzo5UJoNIVkXCofRL1vfAJy4-j1xEwwJPhY20D2LahpmWItiDfYRLVZBtVafViWUy1WSvI2mW%2BXasjAgwVwnp9wuRidGPi1jkvTaJzUebLUl4grDBKyratYLD8MMIjyOozROvjXVV043jWgE4gbMc2TFOG7NTyCEZeKK7gyuq1oJ5M573u%2BxtnP0dzvNns6AvlcLLsXRNipS4HYOy15ZvnAAahqxpxPAcTxBrI7a6LGm67OwP525t1aoI6Ul84vIBWeh2V10Nd15GP3ZyRbt5wb7dG3dyJtT3gWwFgADi2DbbtiD7Ydo-Hb9UznZPEtXU%2BBdU94gjzXiwqitcppYEmdzk4UwXLeQdiHUxXT-nbTtPHZM6Y0c5T2Eq5OEHdg6PTDmKeI5A%2B7kAHjmfocCNpMEtDlWAnRuhYA2odECYQJ4xUuo%2BXS58ZaXwhjAsy5A8HVATAkX%2BQEGBvziJ-Y8BDryjRqq7E%2BpCRKzyDk0QQtMzbMy0DiC4DpqDwFwHuWAB0Ei329pw-Cf0j7ENzmAm6c8tQsjoniTezotCOh6P4AA0ogRA%2BRsF3CwJ0Y0dpEGay1HhIhgM5xKjIYIwu3gWSmzaMafc5AADWOIAi%2BX8l0YUJ5ECM3Ed7Tee0DocMjIA9RwDj561Pt4iBuingslDoE20slECBWjtABgJiBRsXgO2D%2BOZgrQH3HabMR0nhuMbgDEhqwPYswhI4w0ITsBQlkDCaWxs-HFzaLHfpJo7TQGGX0aEuI2iVIdIUeAItuFi14dk6ejVyGTMUCybubRLHWMyrbZhoIgm4FCeEkctED5gH%2BpjFu9Uz4%2BIvicxebRlH31-IgWZXtrgcXyEwXBbSknbxSe0po6TD6ZM0aAlyOihGsmvm0IxVTTG4PQD1SyAQXEdLUl0yIkxWglACLyA6yQ3EzCAA)
That's still valid [code](https://sharplab.io/#v2:DYLgZgzgNALiBOBXAdgH2AUxgAgIbYF5sAGAWAChMcAjQ7GJDCq7AYzoEZmtsATOgEwVuOAPqEK2KfXgBPSdMUNEGbKgB82AJb8N23Zp1rDvKNmoKpAdy0wAFtgwAPbAFp1lxdjC5gEDGY+fhhAA).

For the object instantiation case, the compiler forces parens itself [see](https://sharplab.io/#v2:DYLgZgzgNALiBOBXAdgHxgTwA4FMAEAwgBYCGWMO8AFAJR4C8AsAFB5t4C2OHARpXgDcSwPABUAljGA4QeCDHjjkAczwB7cuLXIGeAKrIAxkRyGA1jgAmAOks4wJRMBhqwAHgD6APjwB3SUR4yjgwUHIhLOyc3HzwgsL68KByCkqq9PpGJuZWtvaOzq6ePv4wgcGh4TAsLNIweIa6xGQUcVSR7BJS+EysUexUAMpqXHgARGN4qD7illMzc9N4szQd/XhQa/16SboTNEA)
I wasn't able to come up with a format, in which the parens could be dropped.